sub/sd_ass.c

```diff
@@ -280,6 +280,14 @@ static int init(struct sd *sd)
     return 0;
 }
 
+// Check if subtitle has events that would cause it to be animated.
+static bool is_animated(char *s)
+{
+    return s && (strstr(s, "\\move") || strstr(s, "\\k") || strstr(s, "\\K") ||
+                 strstr(s, "\\k") || strstr(s, "\\ko") || strstr(s, "\\fad") ||
+                 strstr(s, "\\t"));
```

Comment:
You've got `\k` in here twice, and `\ko` is redundant with it.

Also, this currently makes 7 (and needs to make 5) passes over the string, which could actually be kinda expensive in the case of very large events (and this runs regardless of whether we're in a static-video case or not, right?). Compare what libass does for something similar:

```C
// Return 1 if the event contains tags that will apply overrides the selective
// style override code should not touch. Return 0 otherwise.
int ass_event_has_hard_overrides(char *str)
{
    // look for \pos and \move tags inside {...}
    // mirrors ass_get_next_char, but is faster and doesn't change any global state
    while (*str) {
        if (str[0] == '\\' && str[1] != '\0') {
            str += 2;
        } else if (str[0] == '{') {
            str++;
            while (*str && *str != '}') {
                if (*str == '\\') {
                    char *p = str + 1;
                    if (mystrcmp(&p, "pos") || mystrcmp(&p, "move") ||
                        mystrcmp(&p, "clip") || mystrcmp(&p, "iclip") ||
                        mystrcmp(&p, "org") || mystrcmp(&p, "pbo") ||
                        mystrcmp(&p, "p"))
                        return 1;
                }
                str++;
            }
        } else {
            str++;
        }
    }
    return 0;
}
```

This only runs once per event on demux, so it's probably not a huge deal, but I figured it was reasonable to note.

Reply:
Yeah I copies this from `has_overrides` which is also pretty inefficient. I can definitely make this just loop once.